lib/hcloud/entry_loader.rb

```diff
@@ -86,14 +89,35 @@ def from_response(response, autoload_action: nil)
           return results
         end
 
-        return Action.new(client, action) if attributes.nil? && action
-        return new(client, attributes).tap { |entity| entity.response = response } if action.nil?
+        if action_resp.nil?
+          return new(client, attributes).tap { |entity| entity.response = response }
+        end
 
         [
-          Action.new(client, action),
+          Action.new(client, action_resp),
```

Comment:
```suggestion
          action_resp,
```

shouldn't it be like this, returning an action containing action objects, or did i miss something?

Reply:
You're right. We already get an `Action` object or a list `[Action, ...]` from `_try_load_action`. Will check why this did not result in a test failure. I think actions + resource should be used by `client.servers.create`, so I would expect a test to fail.

Reply:
@tonobo I have analyzed this code path over a few hours :D To my analysis this exact location would only get hit in this situation:

- a method called on an instance type (e.g. `Server`, `Image`, ...; not on a resource type `ServerResource`, ...)
- if the method does not use a block on `prepare_request`
- if the API endpoint that the method calls returns an action AND a matching type for the class (e.g. `server: {}` for `Server` and so on)

I did not find any API endpoint that fulfills these constraints. The closest I could find was `create_image` on Server, which returns an action and an image. Does not fulfill the constraints above (since it returns an image, not a server) and thus takes a different code path. But it comes quite close, so it makes sense to add this test in this PR.